api/access/server.go

```diff
@@ -0,0 +1,506 @@
+	}
+
+	resp := access.BlockResponse{
+		Block: &block,
+	}
+
+	return &resp, nil
+}
+
+func (s *Server) GetCollectionByID(_ context.Context, in *access.GetCollectionByIDRequest) (*access.CollectionResponse, error) {
+	collId := flow.HashToID(in.Id)
+	collection, err := s.index.Collection(collId)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve collection with ID %x: %w", in.Id, err)
+	}
+
+	collEntity := entities.Collection{
+		Id: in.Id,
+	}
+	for _, txID := range collection.Transactions {
+		collEntity.TransactionIds = append(collEntity.TransactionIds, txID[:])
+	}
+
+	resp := access.CollectionResponse{
+		Collection: &collEntity,
+	}
+
+	return &resp, err
+}
+
+func (s *Server) GetTransaction(_ context.Context, in *access.GetTransactionRequest) (*access.TransactionResponse, error) {
+	txID := flow.HashToID(in.Id)
+	tx, err := s.index.Transaction(txID)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve transaction: %w", err)
+	}
+
+	resp := access.TransactionResponse{
+		Transaction: convert.TransactionToMessage(*tx),
+	}
+
+	return &resp, nil
+}
+
+func (s *Server) GetTransactionResult(_ context.Context, in *access.GetTransactionRequest) (*access.TransactionResultResponse, error) {
+	txID := flow.HashToID(in.Id)
+	result, err := s.index.Result(txID)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve transaction result: %w", err)
+	}
+
+	// We also need the height of the transaction we're looking at.
+	// TODO: See if it wouldn't make more sense to remove this index and just
+	//		 always return the status as sealed.
+	//		 https://github.com/optakt/flow-dps/issues/317
+	height, err := s.index.HeightForTransaction(txID)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve block height: %w", err)
+	}
+
+	block, err := s.index.Header(height)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve block header: %w", err)
+	}
+	blockID := block.ID()
+
+	statusCode := uint32(0)
+	if result.ErrorMessage == "" {
+		statusCode = 1
+	}
+
+	status := entities.TransactionStatus_SEALED
+	sealedHeight, err := s.index.Last()
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve last height: %w", err)
+	}
+
+	if height > sealedHeight {
+		status = entities.TransactionStatus_EXECUTED
+	}
+
+	events, err := s.index.Events(height)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve events: %w", err)
+	}
+
+	resp := access.TransactionResultResponse{
+		Status:       status,
+		StatusCode:   statusCode,
+		ErrorMessage: result.ErrorMessage,
+		Events:       convert.EventsToMessages(events),
+		BlockId:      blockID[:],
+	}
+
+	return &resp, nil
+}
+
+func (s *Server) GetAccount(ctx context.Context, in *access.GetAccountRequest) (*access.GetAccountResponse, error) {
+	req := access.GetAccountAtLatestBlockRequest{
+		Address: in.Address,
+	}
+
+	account, err := s.GetAccountAtLatestBlock(ctx, &req)
+	if err != nil {
+		return nil, err
```

Comment:
Because this call is a call to another endpoint, so if we have an error here, it's already wrapped by the `GetAccountAtLatestBlock` endpoint. I don't think it makes sense to wrap the wrapped error by the other endpoint, as how we implement the endpoints in the backend does not matter to the end user, so they shouldn't get a different error depending on the endpoint, IMO

Reply:
Right, that makes sense. Perhaps a comment noting that would be nice, just to let the reader know that it's not forgotten, but deliberate. But no issues really.